src/pks/flow/constitutive_relations/sources/distributed_tiles_evaluator.cc

```diff
@@ -102,8 +124,26 @@ DistributedTilesRateEvaluator::Update_(State& S)
 
     for (AmanziMesh::Entity_ID c = 0; c != ncells; ++c) {
       if (sub_marks[0][c] > 0) {
-        double val = std::min(p_enter_ - pres[0][c], 0.0) * dens[0][c] * k_;
+        if (th_ < 0){
+          //AmanziMesh::Entity_ID_List faces;
+          //std::vector<int> dirs;
+          const auto& [faces, dirs] = mesh->getCellFacesAndDirections(c);
+          double zmax = -1e+98;
+          double zmin =  1e+98;          
+          for (auto f : faces){
+            auto xf = mesh->getFaceCentroid(f);
+            zmax = std::max(zmax, xf[z_index]);
+            zmin = std::min(zmin, xf[z_index]);              
+          }
+          th_ = zmax - zmin;
+          AMANZI_ASSERT(th_ > 1e-12);
+        }
 
+	
+        double diff_p = std::min(p_enter_ - pres[0][c], 0.0); //represents h_average difference 
+        double val = diff_p * dens[0][c] * (12.*kb_*d_)/(L_*L_*visc[0][c] *th_); //linear term [mol/(m^3 s)]
+        val -= 9. * diff_p * diff_p * dens[0][c] * (ka_/ (visc[0][c] * mass_dens[0][c] * L_ * L_ * gr * th_)); // nonlinear term [mol/(m^3 s)]
```

Comment:
please run all code through clang-format -- this can't fit on a single line...

```
cd $AMANZI_SRC_DIR/src
../tools/formatting/clang-format.sh .
```

Reply:
(though probably you should just put the comment on its own line manually)